src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -500,24 +645,6 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id, HR
         return S_OK;
     }
 
-    for (auto&& skip_assembly : skip_assemblies)
```

Comment:
Worth to check if it was not written for the performance purposes.
I suppose it significantly reduces time for well known libraries which should be not instrumented.

Reply:
Let me double-check this.

Reply:
We can't skip redirection for some assemblies in that set, so bailing early on them won't work. That said we don't need to process them in cases when redirection is disabled. We should address the set of skips as an optimization when we [refactor integrations loader](#1482) via #1482. In light of that, I opted to restore the skips for when the redirection is disabled.